src/cache/fetchWithCache.ts

```diff
@@ -0,0 +1,68 @@
+import NodeCache from 'node-cache';
+import { useEffect, useState } from 'react';
+
+// eslint-disable-next-line turbo/no-undeclared-env-vars
+const cacheTTL = Number(process.env.CACHE_TTL) || 600;
```

Comment:
const cacheTTL = Number(process.env.CACHE_TTL) || 600; 
I'm not sure about this, there should be one more guard before consuming the environment variable directly